### PR TITLE
Major Models now randomly generate

### PR DESCRIPTION
**Pull request by tabyington**
Course.random now generates a random course
accounts can be randomly generated
That's right, I put an ascii bob ross in there
Bob was unrecognizable, replaced with the happy little tree.
User and section can now randomly generate

### CODE REVIEW

File: lib/models/account.rb [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
-  @local_dictionaries = File.absolute_path("lib")
-  @row
+  attr_reader :name, :uid, :parent_id, :root_id, :time_zone, :sis_id, :workflow
 
   def to_s
-    string = ""
+    string = "#{@name}, #{@uid}, #{@parent_id}, #{@root_id}, #{@time_zone}, #{@sis_id}, #{@workflow}"
+  end
+
+  def initialize(name, uid, parent, root, time_zone, sis, workflow)
```
Comment (jaxxon4444) [inline lib/models/account.rb]: I'm just wondering if all these arguments should be required as not all of them are necessary for creating an account. Not specifying a parent will default to root. Not specifying a time zone will default to match the root and sis IDs aren't required (though maybe they will be for our needs. But I think we could still let the random method initialize accounts with all these arguments but leave things more flexible for the future by allowing an options hash. No?

File: lib/models/assignment.rb [hunk 2]
```diff
@@ -1,11 +1,24 @@
 class Assignment < Forgery
+  attr_reader :name, :description, :due_at, :lock_at, :course_id, :assignment_group
+  attr_writer :due_at, :lock_at
 
-  def to_s
-    string = ""
+  def initialize(name, description, due_at, lock_at, course, group)
```
Comment (jaxxon4444) [inline lib/models/assignment.rb]: same general question

File: lib/models/account.rb (hunk 1 above)
Comment (tabyington) [reply lib/models/account.rb]: I was fighting with the random stuff early on, so this is the way I was able to get the random init working, but I may have worked out the problems that were causing it. I can mess around with it if you don't like it.

File: lib/models/assignment.rb (hunk 2 above)
Comment (tabyington) [reply lib/models/assignment.rb]: Same

File: lib/models/account.rb (hunk 1 above)
Comment (tabyington) [reply lib/models/account.rb]: I think you may be trying to build in flexibility we won't need.

Comment (tabyington) [reply lib/models/account.rb]: Changes made

**Review by jaxxon4444 — APPROVED** [review]
Approving with the understanding that the opt hashes alone in the initializers is sufficient for now for using the random methods for generation of instances but that eventually we will want to update the initializers to have some required arguments so we can generate instances with specific intent behind values and will be force to send in required values and avoid errors from Canvas.

**Review by jaxxon4444 — APPROVED** [review]
Approving with the understanding that the opt hashes alone in the initializers is sufficient for now for using the random methods for generation of instances but that eventually we will want to update the initializers to have some required arguments so we can generate instances with specific intent behind values and will be force to send in required values and avoid errors from Canvas.

**Pull request merged**